src/ring_theory/polynomial/homogeneous.lean

```diff
@@ -0,0 +1,158 @@
+/-
+Copyleft 2020 Johan Commelin. No rights reserved.
+Authors: Johan Commelin
+-/
+
+import data.mv_polynomial
+import data.fintype.card
+
+open_locale big_operators
+
+namespace mv_polynomial
+variables {σ : Type*} {τ : Type*} {R : Type*} {S : Type*}
+
+--move this
+lemma exists_coeff_ne_zero [comm_semiring R] (φ : mv_polynomial σ R) (h : φ ≠ 0) :
+  ∃ d, coeff d φ ≠ 0 :=
+by { contrapose! h, apply ext, intro d, rw [h d, coeff_zero] }
+
+/-- A multivariate polynomial `φ` is homogeneous of degree `n`
+if all monomials occuring in `φ` have degree `n`. -/
+def is_homogeneous [comm_semiring R] (φ : mv_polynomial σ R) (n : ℕ) :=
+∀ ⦃d⦄, coeff d φ ≠ 0 → ∑ i in d.support, d i = n
```

Comment:
Does it make sense to state this as a `submodule R (mv_polynomial σ R)`?

Reply:
We might want both. This version enables dot notation

Reply:
It looks like this is already merged? Which bits of this PR are left?

Reply:
Probably all the stuff on the various kinds of elementary symmetric polynomials etc...

Reply:
I added the submodule in #6666